### ISSUE TITLE
Getting MSBuild properties into project.json values.

### ISSUE THREAD

**Issue by dls314** [aspnet/Tooling#617]
I'd like to replace a value in project.json at build time with an MSBuild property value.

My best bet so far is to pass the property values to a BeforeBuild MSBuild target and have have that target do the dirty work of modifying the project.json file.

Should that work?
Any terrible gotchas before I get started?
Is there an accepted (easier) way to do this using the current tooling?

**Comment by sayedihashimi** [comment]
@dls314 what properties are you trying to update?

**Comment by dls314** [comment]
I'm trying to update the version property in the project.json with an MSBuild property.

**Comment by barrytang** [comment]
TFS Bug#235751

**Comment by dls314** [comment]
I've run into these issues trying to get down the path of creating an MSBuild task.

- https://github.com/dotnet/cli/issues/1681
- https://github.com/Microsoft/msbuild/issues/692

**Comment by dls314** [comment]
I found that modifying the project.json file on disk before _GetDependencyFragmentFiles runs worked for my case, and then I found that the DOTNET_BUILD_VERSION and DOTNET_ASSEMBLY_FILE_VERSION environment variables were actually the right answer. They can be manipulated later in the build than the general case of modifying project.json. I'm manipulating them in a target before BeforeBuild and it works.
